massa-api/src/public.rs

```diff
@@ -945,6 +957,53 @@ impl MassaRpcServer for API<Public> {
             .collect())
     }
 
+    async fn get_address_datastore_keys(
+        &self,
+        arg: Vec<GetAddressDatastoreKeysRequest>,
+    ) -> RpcResult<Vec<GetAddressDatastoreKeysResponse>> {
+        let requests: Vec<ExecutionQueryRequestItem> = arg
+            .into_iter()
+            .map(|request| {
+                from_get_address_datastore_keys(
```

Comment:
I think that you should directly use `ExecutionController::query_state` with a batched request because:
- it avoids having different requests happen at different slots (if a slot is executed while this loop is running)
- it avoids having to duplicate code